pkg/mutation/schema/node_test.go

```diff
@@ -0,0 +1,253 @@
+package schema
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type idPath struct {
+	types.ID
+	path string
+}
+
+func id(name string) types.ID {
+	return types.ID{Name: name}
+}
+
+func ip(name string, path string) idPath {
+	return idPath{ID: id(name), path: path}
+}
+
+func gvk(group, version, kind string) schema.GroupVersionKind {
+	return schema.GroupVersionKind{Group: group, Version: version, Kind: kind}
+}
+
+func TestNode_Add(t *testing.T) {
+	testCases := []struct {
+		name   string
+		before []idPath
+		add    idPath
+		want   references
+	}{
+		{
+			name:   "no conflict on first add",
```

Comment:
Those sorts of tests are very brittle, and are difficult to read and maintain. It is considered best practice to test the behavior of objects rather than their internals.

I have added a unit test that verifies the behavior you were wondering about.

If this behavior seems too complex to trust to unit tests, it means I have chosen the wrong abstraction and should make a simpler type that can be tested. Let me know if this is the case.

Reply:
In this case we are very interested in making sure the internal state is accurate (particularly after deletes), as otherwise the system could corrupt over time.

Reply:
It doesn't have to test to the same extent as other tests, but we do want to validate that our internal state is working as expected.

Reply:
Ok, I'm adding a couple but making clear that they are structural tests, not intended for verifying behavior.

Reply:
Done.